packages/core/src/endowmentsToolkit.js

```diff
@@ -151,6 +183,81 @@ function endowmentsToolkit({
     return !value
   }
 
+  /**
+   * @param {string} key
+   * @param {Record<string, any>} sourceRef
+   * @param {Record<string, any>} targetRef
+   */
+  function makeWritableValueAtPath(key, sourceRef, targetRef) {
```

Comment:
When we make the assumption of global store being the sourceRef, the implementation becomes very simple.